experimental/PropertyDDS/packages/property-shared-tree-interop/src/schemaConverter.ts

```diff
@@ -0,0 +1,362 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	fail,
+	emptyField,
+	FieldKinds,
+	FieldSchema,
+	NamedTreeSchema,
+	neverTree,
+	rootFieldKey,
+	SchemaData,
+	StoredSchemaRepository,
+	TreeSchemaIdentifier,
+	ValueSchema,
+	lookupTreeSchema,
+	fieldSchema,
+	namedTreeSchema,
+	brand,
+	EmptyKey,
+	TreeTypeSet,
+	TreeType,
+} from "@fluid-internal/tree";
+import { PropertyFactory, PropertyTemplate } from "@fluid-experimental/property-properties";
+import { TypeIdHelper } from "@fluid-experimental/property-changeset";
+
+const booleanTypes = new Set(["Bool"]);
+const numberTypes = new Set([
+	"Int8",
+	"Uint8",
+	"Int16",
+	"Uint16",
+	"Int32",
+	"Int64",
+	"Uint64",
+	"Uint32",
+	"Float32",
+	"Float64",
+	"Enum",
+]);
+const primitiveTypes = new Set([
+	"Bool",
+	"String",
+	"Int8",
+	"Uint8",
+	"Int16",
+	"Uint16",
+	"Int32",
+	"Int64",
+	"Uint64",
+	"Uint32",
+	"Float32",
+	"Float64",
+	"Enum",
+]);
+
+function isIgnoreNestedProperties(typeid: string): boolean {
+	return typeid === "Enum";
+}
+
+function loadInheritedChildren(): Map<string, Set<string>> {
+	const inheritedChildren: Map<string, Set<string>> = new Map();
+	const allTypes = PropertyFactory.listRegisteredTypes();
+	for (const typeid of allTypes) {
+		const parents = PropertyFactory.getAllParentsForTemplate(typeid);
+		for (const parent of parents) {
+			if (!inheritedChildren.has(parent)) {
+				inheritedChildren.set(parent, new Set());
+			}
+			inheritedChildren.get(parent)?.add(typeid);
+		}
+	}
+	return inheritedChildren;
+}
+
+function getInheritedChildrenForType(
+	inheritedChildrenByType: Map<string, ReadonlySet<string>>,
+	typeid: string,
+): ReadonlySet<TreeSchemaIdentifier> {
+	return getInheritedChildrenForTypes(inheritedChildrenByType, new Set([typeid]));
+}
+
+function getInheritedChildrenForTypes(
+	inheritedChildrenByType: Map<string, ReadonlySet<string>>,
+	types: ReadonlySet<string>,
+): ReadonlySet<TreeSchemaIdentifier> {
+	const result = new Set<TreeSchemaIdentifier>();
+	for (const type of types) {
+		result.add(brand(type));
+		const strSet = inheritedChildrenByType.get(type) ?? new Set();
+		for (const str of strSet) {
+			result.add(brand(str));
+		}
+	}
+	return result;
+}
+
+export function convertPSetSchemaToSharedTreeLls(
+	repository: StoredSchemaRepository,
+	rootFieldSchema: FieldSchema,
+): void {
+	const inheritedChildrenByType = loadInheritedChildren();
+	const globalTreeSchema: Map<TreeSchemaIdentifier, NamedTreeSchema> = new Map();
+	// Extract all referenced typeids for the schema
+	const unprocessedTypeIds: TreeSchemaIdentifier[] = [];
+	const rootBaseTypes = rootFieldSchema.types ?? fail("Expected root types");
```

Comment:
This only registers types from the root field ot the document and those that are referenced via chain starting at the root. Should we provide a mechanism to register additional types, which will then be available, e.g. to add them to a NodeProperty, without being mentioned in the schema?